privval/priv_validator.go

```diff
@@ -142,19 +195,19 @@ func (pv *FilePV) save() {
 // NOTE: Unsafe!
 func (pv *FilePV) Reset() {
 	var sig []byte
-	pv.LastHeight = 0
-	pv.LastRound = 0
-	pv.LastStep = 0
-	pv.LastSignature = sig
-	pv.LastSignBytes = nil
+	pv.LastSignState.Height = 0
+	pv.LastSignState.Round = 0
+	pv.LastSignState.Step = 0
+	pv.LastSignState.Signature = sig
+	pv.LastSignState.SignBytes = nil
 	pv.Save()
 }
 
 // SignVote signs a canonical representation of the vote, along with the
 // chainID. Implements PrivValidator.
 func (pv *FilePV) SignVote(chainID string, vote *types.Vote) error {
-	pv.mtx.Lock()
-	defer pv.mtx.Unlock()
+	pv.LastSignState.mtx.Lock()
```

Comment:
Maybe we can move lock & unlock into `saveSigned` function? do we need to hold the lock during the whole signing process?

Reply:
maybe there is concern about concurrent signing